src/main/date.js

```diff
@@ -0,0 +1,30 @@
+import moment from 'moment'
+import times from '../extras/times'
+
+function date(arr) {
+  let i = 0
+  const regex = /(\w+)[\#](\w+)[\#](\w+)/ig
+  const regexStar = /(\w+)\*(\w+)\*(\w+)/ig
+  while ( i < 1000) {
```

Comment:
I wouldnt hard code a number into a while loop. I think finding a good invariant would be more general and would allow you to work with a data set grater than 1000.